src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -2362,4 +2398,230 @@ bool TopoShape::fixSolidOrientation()
+                }
+                expandCompound(s, _shapes);
+            }
+            else if (_shapes.size()) {
+                _shapes.push_back(s);
+            }
+        }
+    }
+
+    if (tolerance > 0.0 && _shapes.empty()) {
+        _shapes = shapes;
+    }
+
+    const auto& inputs = _shapes.size() ? _shapes : shapes;
+    if (inputs.empty()) {
+        FC_THROWM(NullShapeException, "Null input shape");
+    }
+    if (inputs.size() == 1) {
+        *this = inputs[0];
+        if (shapes.size() == 1) {
+            // _shapes has fewer items than shapes due to compound expansion.
+            // Only warn if the caller paseses one shape.
+            FC_WARN("Boolean operation with only one shape input");
+        }
+        return *this;
+    }
+
+    std::unique_ptr<BRepAlgoAPI_BooleanOperation> mk;
+    if (strcmp(maker, Part::OpCodes::Fuse) == 0) {
+        mk.reset(new BRepAlgoAPI_Fuse);
+    }
+    else if (strcmp(maker, Part::OpCodes::Cut) == 0) {
+        mk.reset(new BRepAlgoAPI_Cut);
+    }
+    else if (strcmp(maker, Part::OpCodes::Common) == 0) {
+        mk.reset(new BRepAlgoAPI_Common);
+    }
+    else if (strcmp(maker, Part::OpCodes::Section) == 0) {
+        mk.reset(new BRepAlgoAPI_Section);
+        buildShell = false;
+    }
+    else {
+        FC_THROWM(Base::CADKernelError, "Unknown maker");
+    }
+
+    TopTools_ListOfShape shapeArguments, shapeTools;
+
+    int i = -1;
+    for (const auto& shape : inputs) {
+        if (shape.isNull()) {
+            FC_THROWM(NullShapeException, "Null input shape");
+        }
+        if (++i == 0) {
+            shapeArguments.Append(shape.getShape());
+        }
+        else if (tolerance > 0.0) {
+            auto& s = _shapes[i];
+            // workaround for http://dev.opencascade.org/index.php?q=node/1056#comment-520
+            s.setShape(BRepBuilderAPI_Copy(s.getShape()).Shape(), false);
+            shapeTools.Append(s.getShape());
+        }
+        else {
+            shapeTools.Append(shape.getShape());
+        }
+    }
+
+#if OCC_VERSION_HEX >= 0x070500
+    // -1/22/2024 Removing the parameter.
+    // if (PartParams::getParallelRunThreshold() > 0) {
+        mk->SetRunParallel(Standard_True);
+        OSD_Parallel::SetUseOcctThreads(Standard_True);
+    // }
+#else
+    // 01/22/2024 This will be an extremely rare case, since we don't
+    // build against OCCT versions this old.  Removing the parameter.
+    mk->SetRunParallel(true);
+    // Only run parallel
+    // if (shapeArguments.Size() + shapeTools.Size() > 2) {
+    //     mk->SetRunParallel(true);
+    // }
+    // else if (PartParams::getParallelRunThreshold() > 0) {
+    //     int total = 0;
+    //     for (const auto& shape : inputs) {
+    //         total += shape.countSubShapes(TopAbs_FACE);
+    //         if (total > PartParams::getParallelRunThreshold()) {
+    //             mk->SetRunParallel(true);
+    //             break;
+    //         }
+    //     }
+    // }
```

Comment:
Why leave the commented-out block?

Reply:
At the time it was to call attention to code not being transferred in.  It's gone now.  Thanks!